OpenDreamRuntime/Objects/DreamObject.cs

```diff
@@ -403,6 +403,32 @@ public class DreamObject {
             throw new InvalidOperationException($"Multiplication cannot be done between {this} and {b}");
         }
 
+        // *=
+        public virtual ProcStatus OperatorMultiplyRef(DreamValue b, DMProcState state, out DreamValue result, in DreamReference reference) {
+            if(TryGetProc("operator*=", out var proc)) {
+                if(!AssignRefProcState.Pool.TryPop(out var operatorProcState)){
+                    operatorProcState = new AssignRefProcState();
+                }
+
+                operatorProcState.Initialize(state, proc, this, state.Usr, new DreamProcArguments(b), reference);
+                state.Thread.PushProcState(operatorProcState);
+                result = DreamValue.Null;
+                return ProcStatus.Called;
+            } else if(TryGetProc("operator*", out var multiplyProc)) {
+                if(!AssignRefProcState.Pool.TryPop(out var operatorProcState)){
+                    operatorProcState = new AssignRefProcState();
+                }
+
+                operatorProcState.Initialize(state, multiplyProc, this, state.Usr, new DreamProcArguments(b), reference);
+                state.Thread.PushProcState(operatorProcState);
+                result = DreamValue.Null;
+                return ProcStatus.Called;
+            } else {
+                throw new InvalidOperationException($"Multiplication cannot be done between {this} and {b}");
+            }
```

Comment:
Cut down on the duplicated code here by doing both `TryGetProc()`s in the same `if`
```suggestion
            if (TryGetProc("operator*=", out var proc) || TryGetProc("operator*", out proc)) {
                if(!AssignRefProcState.Pool.TryPop(out var operatorProcState)){
                    operatorProcState = new AssignRefProcState();
                }

                operatorProcState.Initialize(state, proc, this, state.Usr, new DreamProcArguments(b), reference);
                state.Thread.PushProcState(operatorProcState);
                result = DreamValue.Null;
                return ProcStatus.Called;
            }

            throw new InvalidOperationException($"Multiplication cannot be done between {this} and {b}");
```